File: wk2/pset2/readability/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>

int main(void)
{
    string text = get_string("Text: ");
    float s = 0;
    float w = 1;
    float l = 0;
    int i = 0;
    char c;
    do
    {
        c = text[i];
        int j = c;
        i++;
        if (c == '.' || c == '!' || c == '?')
        {
            s++;
        }
        else if (c == ' ')
        {
            w++;
        }
        else if ((64 < j && j < 91) || (96 < j && j < 123))
        {
            l++;
        }
    }
    while (c != '\0');
    // printf("s: %f w: %f l: %f\n", s, w, l);
    float L = (float) (l / w) * 100;
    float S = (double) (s / w) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        int ans = round(index);
        if (ans >= 16)
        {
            printf("Grade 16+\n");
        }
        else
        {
            printf("Grade %i\n", ans);
        }
    }
}

File: wk6/pset/sentimental-readability/readability.py
from cs50 import get_string

text = get_string("Text: ")

s = 0.0  # sentences
w = 1.0  # words
l = 0.0  # letters
c = ''

for i in range(len(text)):
    # get the current letter
    c = text[i]
    # save the ascii value for later
    j = ord(c.lower())
    # if this is the end of a sentence add one to sentences
    if c == '.' or c == '!' or c == '?':
        s += 1
    # if this is the end of a word add one to words
    elif c == ' ':
        w += 1
    # if this is a letter add one to letters
    elif 96 < j and j < 123:
        l += 1

L = l / w * 100
S = s / w * 100
index = 0.0588 * L - 0.296 * S - 15.8
if index < 1:
    print("Before Grade 1")
else:
    ans = round(index)
    if ans >= 16:
        print("Grade 16+")
    else:
        print("Grade", ans)
